File: py/3.py
# 6857
import math
import euler
m = 0
N = 600851475143
sieve = euler.prime_sieve(int(math.sqrt(N)))
for p in xrange(len(sieve)):
    if sieve[p] and N % p == 0:
        m = p
print m

File: c/3.c
/* 6857 */
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include "euler.h"

#define N 600851475143LL

int main(int argc, char **argv)
{
    int i, np;
    char *sieve;
    int *primes;

    np = prime_sieve(sqrt(N), &sieve, &primes);
    for (i = np - 1; i > 0 ; i--)
    {
        if (N % primes[i] == 0)
        {
            printf("%d\n", primes[i]);
            break;
        }
    }
    free(sieve);
    free(primes);
    return 0;
}
